pinot-common/src/main/java/org/apache/pinot/common/function/FunctionUtils.java

```diff
@@ -69,6 +80,12 @@ private FunctionUtils() {
     put(Double.class, DataType.DOUBLE);
     put(String.class, DataType.STRING);
     put(byte[].class, DataType.BYTES);
+    put(int[].class, DataType.INT);
+    put(String[].class, DataType.STRING);
+    put(long[].class, DataType.LONG);
+    put(float[].class, DataType.FLOAT);
+    put(double[].class, DataType.DOUBLE);
+    put(boolean.class, DataType.BOOLEAN);
```

Comment:
Don't add `boolean` here as `DataType.BOOLEAN` is not a valid internal `DataType` (we use `STRING` to represent boolean). All the unrecognized types will be handled as `STRING`

Reply:
boolean been removed